### PR TITLE
Decouple tests on routing/basic_test

### PR DESCRIPTION
**Pull request by cibernox**
I'm copying one by one the tests in `packages/ember/tests/routing/basic_test.js` to `packages/ember/tests/routing/decoupled_basic_test.js` one by one, and updating the syntax.

This is an ongoing PR, but feel free to merge anytime. If this PR is merged I'll continue working on a new one, so there is no reason to hold this back.

### CODE REVIEW

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 1]
```diff
@@ -710,4 +749,1064 @@ class extends ApplicationTestCase {
...
+      let router = this.applicationInstance.lookup('router:main');
+      this.handleURL(assert, '/foo/bar/baz');
+      assert.equal(applicationController.get('currentPath'), 'foo.bar.baz');
+      run(() => router.send('goToQux'));
+      assert.equal(applicationController.get('currentPath'), 'foo.qux');
+      assert.equal(router.get('location').getURL(), '/foo/qux');
+    });
+  }
+
+  ['@test Generated names can be customized when providing routes with dot notation'](assert) {
```
Comment (mixonic) [inline packages/ember/tests/routing/decoupled_basic_test.js]: lol what is this test title? I see that it is matches the previous test, but I have no idea where "dot notation" is used in this test.

**Review by mixonic — CHANGES_REQUESTED** [review]
@cibernox Thank you for your continued work on this. Improving the test suite unblocks feature work like async rendering and cleanup like removing the default globals resolver.

The big change I'm going to ask for is to refactor use of `handleUrl` and the helper methods in to `visit` based calls. AFAICT, none of these tests are `handleURL` specific. Instead, they used that API because the predated our public-facing `visit` API for navigating an Ember app.

Additionally some of the tests ported in this PR do:

```js
return this.visit('/').then(() => {
  run(() => this.handlUrl(assert, somePath);
  // more assertions
});
```

In these cases the "visit then handle" is probably not needed. If the test is not specifically asserting the behavior of a transition (from / to the handled route) these tests can just become:

```js
return this.visit(somePath).then(() => {
  // more assertions
});
```

Again the old tests were likely booting then handling a URL not because they wanted a transition, but because they could not boot the application with an initial path besides `/`.

File: packages/ember/tests/routing/decoupled_basic_test.js [hunk 2]
```diff
@@ -710,4 +749,1064 @@ class extends ApplicationTestCase {
...
+
+    this.add('route:bar-baz', Route.extend({
+      enter() {
+        assert.ok(false, 'Should abort transition getting to next route');
+      }
+    }));
+
+    return this.visit('/').then(() => {
+      let router = this.applicationInstance.lookup('router:main');
+      this.handleURLAborts(assert, '/foo/bar/baz');
```
[the remark this answers is not in the document]
Comment (cibernox) [reply packages/ember/tests/routing/decoupled_basic_test.js]: I couldn't find a way of implementing that method on top of `visit`. Apparently it never rejects. Is it ok to implement it on top of `router.handleURL`?

[the remark this answers is not in the document]
Comment (mixonic) [reply packages/ember/tests/routing/decoupled_basic_test.js]: Hm. We would need to use an error substate to reflect the failure for assertion.

Yeah these changes are an improvement as-is. I'm sure later we will need to visit this synchronous routing.

**Review by mixonic — APPROVED** [review]

**Comment by cibernox** [comment]
@mixonic Can you merge this? I'll continue in another branch.

**Comment by mixonic** [comment]
@cibernox done! Thanks.

If you can please squash down edits like `Apply feedback` in the future that would be helpful to keep our repo history readable. 🍻

**Pull request merged**